### ISSUE TITLE
gpg signing error building .deb from source

### ISSUE THREAD

**Issue by parke** [stefansundin/truecrypt.deb#1]
I followed the instructions in `README.md` for building the `.deb` package from source.  (I'm installing on a system running a beta of Ubuntu 15.10.)

When I ran `debuild` I got the following error:

```
Now signing changes and any dsc files...
 signfile truecrypt_7.1a-8.dsc Stefan Sundin <stefan@stefansundin.com>
gpg: directory `/home/bake/.gnupg' created
gpg: new configuration file `/home/bake/.gnupg/gpg.conf' created
gpg: WARNING: options in `/home/bake/.gnupg/gpg.conf' are not yet active during this run
gpg: keyring `/home/bake/.gnupg/secring.gpg' created
gpg: keyring `/home/bake/.gnupg/pubring.gpg' created
gpg: skipped "Stefan Sundin <stefan@stefansundin.com>": secret key not available
gpg: /tmp/debsign.wANSEYaR/truecrypt_7.1a-8.dsc: clearsign failed: secret key not available
debsign: gpg error occurred!  Aborting....
debuild: fatal error at line 1293:
running debsign failed
```

I imagine that this error is expected, as third parties such as myself will not have access to the secret key used to sign the packages.

You might want to document that this error will happen and is expected.

Or perhaps(?) you could update your build scripts so that signing is done separately.  Then third parties such as myself would be able to build truecrypt without any errors.

Cheers!

**Comment by stefansundin** [comment]
Ah.. yes. Sorry about that. Running just `debuild` will try to sign as me (as specified in the Maintainer field in the control file).

It would appear that the command that you are looking for is:

```
debuild -i -us -uc -b
```

This should build a binary file without trying to sign anything. I will update the README file. Thanks for reporting this.

Also see these manpages:
- http://man.he.net/man1/debuild
- http://man.he.net/man1/dpkg-buildpackage

This truecrypt package was my first debian package, and it took some doing to build it successfully. I wish it was easier to get into, but it is what it is. I hope this helps you.

An alternative way to try this on 15.10 is to download the deb file from launchpad (e.g. this file: https://launchpad.net/~stefansundin/+archive/ubuntu/truecrypt/+files/truecrypt_7.1a-8_amd64.deb), and then double-click it.

**Comment by stefansundin** [comment]
It's funny. I think I just wrote down the exact steps in the README file that I run every time I need to test the build on a new version of Ubuntu. 6 months is just enough time to forget how things work. You might be the first one besides me from following them.

**Comment by parke** [comment]
> This truecrypt package was my first debian package, and it took some doing to build it successfully. I wish it was easier to get into, but it is what it is. I hope this helps you.

I have tried on at least two occasions, maybe more, to learn how to make Debian packages.  Each time, the process got too complicated, and I abandoned the pursuit.  There are just too many different programs, files and configuration files involved.  The last time was in December of 2014, and I decided I would never try again.  Life is too short.  So I certainly appreciate the work of people who are willing to learn how to jump through the excessive number of hoops.

Regarding using the existing binary package... I was not sure whether or not the existing packages would work across kernel versions.

Cheers!

**Comment by parke** [comment]
Just FYI:  There appear to be problems with at least some TrueCrypt functionality on Ubuntu 15.10.  I get "Invalid characters encountered" errors when trying to mount a system-encrypted TrueCrypt volume.  I've found another user reporting this problem, and claiming it might be due to incompatibilities with newer versions of Fuse.

https://bugzilla.opensuse.org/show_bug.cgi?id=939435

http://veracrypt2.rssing.com/chan-19067982/all_p117.html

As a workaround, I have installed TrueCrypt on an Ubuntu 14.04 system, using your PPA.  On 14.04, TrueCrypt is able to mount and read the encrypted volume.  This is good enough for me, as I am just trying to export legacy data off of TrueCrypt.  I do not expect to conduct additional testing on 15.10.

Cheers!
